### PR TITLE
Eng 8959

### PR DESCRIPTION
**Pull request by manjujames**
Process update catalog requests only if server is ready.

### CODE REVIEW

**Comment by pdshaw** [comment]
Linux build fails with the pro-master.

Maybe you need to merge some recent change that matches on the Voltdb side?

I can check a little later since you'll probably be on the road for a while.

On Fri, Sep 4, 2015 at 6:05 PM, manjujames <notifications@github.com> wrote:

> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/VoltDB/voltdb/pull/2790
> Commit Summary
>
>    - Flag that server has started and apply catalog updates only if
>    server is up.
>    - ENG-8959: Removing debug stmts.
>
> File Changes
>
>    - *M* src/frontend/org/voltdb/importer/ImportManager.java
>    <https://github.com/VoltDB/voltdb/pull/2790/files#diff-0> (13)
>
> Patch Links:
>
>    - https://github.com/VoltDB/voltdb/pull/2790.patch
>    - https://github.com/VoltDB/voltdb/pull/2790.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/VoltDB/voltdb/pull/2790>.
>



-- 
*Peter Shaw,* VoltDB Engineering
*VoltDB, Inc.*
*E:* pshaw@voltdb.com

www.voltdb.com <http://voltdb.com>
Follow VoltDB on Twitter <https://twitter.com/voltdb>, Facebook
<https://www.facebook.com/VoltDB?sk=wall>, and LinkedIn
<http://www.linkedin.com/company/voltdb>

File: src/frontend/org/voltdb/importer/ImportManager.java [hunk 1]
```diff
@@ -69,6 +69,8 @@
     private final int m_myHostId;
     private Framework m_framework;
     private ChannelDistributer m_distributer;
+    private boolean serverStarted;
```
Comment (akhanzode) [inline src/frontend/org/voltdb/importer/ImportManager.java]: m_serverStarted

File: src/frontend/org/voltdb/importer/ImportManager.java [hunk 2]
```diff
@@ -208,6 +210,15 @@ public void updateCatalog(CatalogContext catalogContext, HostMessenger messenger
     }
 
     public synchronized void readyForData(CatalogContext catalogContext, HostMessenger messenger) {
+        serverStarted = true; // Note that server is ready, so that we know whether to process catalog updates
+        readyForDataInternal(catalogContext, messenger);
+    }
+
+    public synchronized void readyForDataInternal(CatalogContext catalogContext, HostMessenger messenger) {
+        if (!serverStarted) {
+            return;
```
Comment (akhanzode) [inline src/frontend/org/voltdb/importer/ImportManager.java]: maybe add a debug log?

**Comment by manjujames** [comment]
Anish, your feedback has been incorporated.

Peter, I merged with master to pick up Anish's change, so this branch is now up-to-date with master.

**Pull request closed without merging**